rust/lance/src/io/exec/fts.rs

```diff
@@ -178,11 +175,12 @@ impl ExecutionPlan for FtsExec {
                             Arc::new(Float32Array::from(scores)),
                         ],
                     )?;
-
-                    Ok::<_, DataFusionError>(batch)
+                    let indexed_stream = stream::iter(vec![Ok(batch)]);
+                    Ok::<_, DataFusionError>(indexed_stream.chain(unindexed_result_stream))
```

Comment:
It would be preferable to express this in terms of a datafusion plan, so we can make this more composable. It would look like:

```
UnionExec
    FtsIndexExec
    FtsFlatExec
        Scan(fragments=new)
```

Reply:
fixed